client/state/utils/serialize.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import { getInitialState } from '@automattic/state-utils';
+import type { Reducer, AnyAction } from 'redux';
+
+export type SerializableReducer< TState, TAction extends AnyAction > = Reducer<
+	TState,
+	TAction
+> & {
+	serialize?: ( state: TState ) => any;
+	deserialize?: ( persisted: any ) => TState;
+};
+
+export function serialize< TState, TAction extends AnyAction >(
+	reducer: SerializableReducer< TState, TAction >,
+	state: TState
+): any {
```

Comment:
What does `serialize` return/do?

Reply:
It returns a serialized state object. This might typically mean it turns it into a string? I'm not sure.

Reply:
The return value is a generic object. One that can be serialized into a string either by `JSON.stringify` or by the structured clone algorithm that is used by IndexedDB to store values.

There are certain constraints on what such an object can contain (no custom class instances, no loops, ...) but these can't be described as a type constraint.

The native APIs typically type these values as `any`.